components/collector/src/source_collectors/gatling/base.py

```diff
@@ -1,44 +1,35 @@
 """Base classes for Gatling collectors."""
 
-import re
-from typing import ClassVar
+from abc import ABC
+from typing import cast
 
-from base_collectors import FileSourceCollector, JSONFileSourceCollector, SourceCollector
-from collector_utilities.type import URL
-from model import SourceResponses
+from bs4 import Tag
 
+from base_collectors import HTMLFileSourceCollector
+from collector_utilities.type import URL, Response
 
-class GatlingCollector(SourceCollector):
-    """Base class for Gatling collectors."""
+
+class GatlingHTMLCollector(HTMLFileSourceCollector, ABC):
+    """Base class for Gatling HTML collectors."""
 
     INDEX_HTML = "/index.html"
-    FILEPATH = "subclass responsibility"
 
     async def _api_url(self) -> URL:
-        """Extend to translate the HTML URL into a URL to the Gatling file that the collector needs."""
+        """Extend to translate the HTML URL into a URL that the collector needs."""
         api_url = await super()._api_url()
-        if api_url.endswith(self.INDEX_HTML):
-            return URL(api_url.removesuffix(self.INDEX_HTML) + self.FILEPATH)
-        return api_url
-
-
-class GatlingJSONCollector(JSONFileSourceCollector, GatlingCollector):
-    """Base class for Gatling collectors that read the stats.json file."""
-
-    FILEPATH = "/js/stats.json"
-
-
-class GatlingLogCollector(FileSourceCollector, GatlingCollector):
-    """Base class for Gatling collectors that read the simulation.log file."""
-
-    file_extensions: ClassVar[list[str]] = ["log"]
-    FILEPATH = "/simulation.log"
-
-    @classmethod
-    async def _timestamps(cls, responses: SourceResponses) -> set[int]:
-        """Return the timestamps in the simulation.log."""
-        timestamps = set()
-        for response in responses:
-            text = await response.text()
-            timestamps |= {int(timestamp) for timestamp in re.findall(r"\d{13}", text)}
-        return timestamps
+        return api_url if api_url.endswith(self.INDEX_HTML) else URL(api_url + self.INDEX_HTML)
+
+    async def simulation_information(self, response: Response, label: str) -> str:
+        """Return the simulation information labeled with the label."""
+        soup = await self._soup(response)
+        label_span = soup.find("span", {"class": "simulation-information-label"}, string=f"{label}: ")
+        if label_span and (parent := label_span.parent):
+            return str(parent.find_all("span")[1].get_text())  # The second span contains the information
+        return ""
+
```

Comment:
These two methods make very strong assumptions about the way the data in HTML is structured. I think it might be worth documenting it in code somewhere, but I'm not sure as to the best location. Perhaps a mention in the docstring that the data lookups are strongly based on assumptions of HTML structure?